utils/zerovec/src/hashmap/serde.rs

```diff
@@ -71,6 +71,10 @@ mod test {
     ];
 
     #[derive(Serialize, Deserialize)]
+    #[allow(
+        dead_code,
+        reason = "Remove this if the lint behavior changes. The dead_code lint explicitly ignores constructing trait impls."
```

Comment:
```suggestion
        reason = "The dead_code lint explicitly ignores constructing trait impls."
```

Reply:
I guess this isn't necessary now that it's not an expect

(I like to distinguish between expects where it is a bug for the lint to not hit vs where it is a signal the directive must be removed)